src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -110,8 +115,12 @@ public final class StringConcatFactory {
      * While the maximum number of argument slots that indy call can handle is 253,
      * we do not use all those slots, to let the strategies with MethodHandle
      * combinators to use some arguments.
+     *
+     * @since 21
      */
-    private static final int MAX_INDY_CONCAT_ARG_SLOTS = 200;
+    @PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
+    public static final int MAX_INDY_CONCAT_ARG_SLOTS;
+    static { MAX_INDY_CONCAT_ARG_SLOTS = 200; }
```

Comment:
I think this deserve a comment such as "use static initialize block to avoid MAX_INDY_CONCAT_ARG_SLOTS being treating as a constant for constant folding," assuming that is the intention here.